modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/HeapLockManager.java

```diff
@@ -282,6 +298,14 @@ private LockState lockState(LockKey key) {
                 if (v == null) {
                     res[0] = slots[index];
                     assert !res[0].markedForRemove;
+
+                    long currentTs = coarseCurrentTimeMillis();
+
+                    long previousTs = lastTsPintWarn.get();
+
+                    if ((currentTs - previousTs) > warningPrintInterval && lastTsPintWarn.compareAndSet(previousTs, currentTs)) {
```

Comment:
Why did you implement it that way? I was expecting that you will log warnin every time when log manager runs out of slots. And when some slots will be released you will reset printWarn flag.

Reply:
If we do not pay attention to the enormous amount of locks, we will be able to see the log file. But it might not be possible if we log it only once and the file is rotated.

Reply:
I'd expect that we will release some locks on conflicts and thus cleanup some slots.